tests/manager/manager-ctor_error-003.phpt

```diff
@@ -201,18 +200,6 @@ Expected string for "authSource" URI option, array given
 OK: Got MongoDB\Driver\Exception\InvalidArgumentException
 Expected string for "authSource" URI option, document given
 OK: Got MongoDB\Driver\Exception\InvalidArgumentException
-Expected string for "gssapiServiceName" URI option, boolean given
-OK: Got MongoDB\Driver\Exception\InvalidArgumentException
-Expected string for "gssapiServiceName" URI option, double given
-OK: Got MongoDB\Driver\Exception\InvalidArgumentException
-Expected string for "gssapiServiceName" URI option, 32-bit integer given
-OK: Got MongoDB\Driver\Exception\InvalidArgumentException
-Expected string for "gssapiServiceName" URI option, ObjectId given
-OK: Got MongoDB\Driver\Exception\InvalidArgumentException
-Expected string for "gssapiServiceName" URI option, array given
-OK: Got MongoDB\Driver\Exception\InvalidArgumentException
-Expected string for "gssapiServiceName" URI option, document given
-OK: Got MongoDB\Driver\Exception\InvalidArgumentException
```

Comment:
I understand that this was removed because `mongoc_uri_option_is_utf8()` no longer reports `true` for "gssapiServiceName" (see: https://github.com/mongodb/mongo-c-driver/commit/4e8848841e826ac1934fa854b5a14a91d5eb118c#diff-c1dfae86645148a1b148cd418f1a52f1). However, I'm concerned that we now might not be handling the option at all.

`php_phongo_apply_options_to_uri()` may need a new condition that specifically checks for an array option matching `MONGOC_URI_GSSAPISERVICENAME` and sets it on the libmongoc URI using the same logic in the aforementioned commit.

Reply:
That's unfortunately not too easy, as the logic you point to uses ``mongoc_uri_parse_auth_mechanism_properties`` which is an internal function. I can of course copy all of it from ``src/libmongoc/src/mongoc/mongoc-uri.c``, but that's not great. Can you think of anything better?

Reply:
We do have `mongoc_uri_set_mechanism_properties()`, which works off of a `bson_t *` and is likely even easier to work with.

In `php_phongo_apply_options_to_uri()`, I expect you'll want to add a case for `MONGOC_URI_GSSAPISERVICENAME` that then creates a new BSON document (e.g. `{ "SERVICE_NAME": "user-supplied-value" }`. For consistency with https://github.com/mongodb/mongo-c-driver/commit/4e8848841e826ac1934fa854b5a14a91d5eb118c, you'll first want to check that there are no authMechanismProperties assigned before setting them from "gssapiServiceName". If `mongoc_uri_get_mechanism_properties()` returns `true`, then we have a warning condition (e.g. "authMechanismProperties SERVICE_NAME already set, ignoring '%s'"). Perhaps that means we do nothing, as I don't think the libmongoc warning would result in a PHP exception. Since PHPC doesn't provide any insight into libmongoc's URI struct (certainly not to PHP userland), I imagine we can't test this no-op behavior (i.e. "gssapiServiceName" won't override an "authMechanismProperties" option).

If `mongoc_uri_get_mechanism_properties()` returns `false`, we should then assert that the "gssapiServiceName" option is a string type. If not, we can continue to raise the unexpected type exceptions. Let's restore the original contents of the `manager-ctor_error-003.phpt` test.

If we do have a string, we can proceed with creating the BSON document from the "gssapiServiceName" value and assigning it with `mongoc_uri_set_mechanism_properties()`.

Reply:
Done